corehq/util/workbook_reading/adapters/xlsx.py

```diff
@@ -59,13 +60,27 @@ def _make_cell_value(self, cell):
                 return cell.value
         return cell.value
 
-    def iter_rows(self):
+    @property
+    @memoized
+    def _max_row(self):
+        # self._worksheet.max_row is the max_row with data OR WITH FORMATTING
+        # That means that if formatting is applied, that will be the xlsx row limit
+        # Note that this is 1-indexed for consistency with openpyxl
+        max_row = 1
         for row in self._worksheet.iter_rows():
+            if any(cell.value for cell in row):
+                max_row = row[0].row
+        return max_row
+
+    def iter_rows(self):
+        for i, row in enumerate(self._worksheet.iter_rows(), 1):
+            if i > self._max_row:
+                break
             yield [Cell(self._make_cell_value(cell)) for cell in row]
 
     def to_worksheet(self):
         # Note that an empty sheet and a sheet with one row both have max_row = 1
-        return Worksheet(title=self._worksheet.title, max_row=self._worksheet.max_row,
```

Comment:
Should this be updated as well?

https://github.com/dimagi/commcare-hq/blob/57f56f1c5ab1243126057ad5eeb3c5fe7a1868e9/corehq/apps/case_importer/util.py#L127-L129

Reply:
I don't think so, I believe `self._worksheet` there is created here:

https://github.com/dimagi/commcare-hq/blob/c6155325492cdf45708085351907aac952a73a7c/corehq/util/workbook_reading/adapters/xlsx.py#L81-L84
via 
https://github.com/dimagi/commcare-hq/blob/c6155325492cdf45708085351907aac952a73a7c/corehq/util/workbook_reading/adapters/generic.py#L9-L19